ledger/apply/asset.go

```diff
@@ -100,12 +100,7 @@ func AssetConfig(cc transactions.AssetConfigTxnFields, header transactions.Heade
 			return err
 		}
 
-		// Record the index used. No separate config for activating
-		// storage in AD because inner transactions can't be turned on
-		// without this change.
-		if balances.ConsensusParams().MaxInnerTransactions > 0 {
-			ad.ConfigAsset = newidx
-		}
+		ad.ConfigAsset = newidx
```

Comment:
It's useful for the value to be in the ApplyData, where it can be accessed by later transactions that use `itxn CreatedConfigID`. And, in fact, I changed `gaid` so that it uses this mechanism to get a created id as well. (Rather than track created ids in two different ways.). `gaid` exists even before we had inner transactions, so this must be stored for its use regardless of whether MaxInnerTransactions is turned on.

However, the ApplyData that we write in old protocols does not include this. It is cleared out the same way we clear out rewards info before the AD was supposed to have it.